homeassistant/components/broadlink/switch.py

```diff
@@ -195,6 +195,9 @@ def __init__(self, device, config):
             device, config.get(CONF_COMMAND_ON), config.get(CONF_COMMAND_OFF)
         )
         self._attr_name = config[CONF_NAME]
+        self._attr_unique_id = (
+            f"{self._device.unique_id}-{self._attr_name.lower().replace(' ', '_')}"
+        )
```

Comment:
This is breaking for existing users, can we migrate existing entities?

Additionally, the name cannot be used, as it user-changeable, does not viable for use in unique IDs.

Reply:
I may be wrong but, looking at the BroadlinkRMSwitch class from before 2021.7b0, it doesn't look like it previously assigned unique ids for the entities, they were added in https://github.com/home-assistant/core/pull/52177 so this looks like it may only be a breaking change if you are on the beta.

Since these are YAML configured would it actually be better to remove the unique id (but that would still be a breaking change if you're on the beta)?

Reply:
And, if we want unique ids for these, we could add a separate **unique_id** option for the switch (like the option for the **template** integration).

Reply:
You are right about the *unique_id* added in #52177, however it looks like an error, from what I see in moving this line will put things back to how they were before:
https://github.com/home-assistant/core/blob/4abdeec36ddcf7d69c855df50bec375b4efc3bd6/homeassistant/components/broadlink/switch.py#L149
to below:
https://github.com/home-assistant/core/blob/4abdeec36ddcf7d69c855df50bec375b4efc3bd6/homeassistant/components/broadlink/switch.py#L217
Without any breaking changes for those who are not on dev

Reply:
Yup, that indeed a regression issue, well spotted @thecode.
Will write up a quick PR to at least restore the previous functionality with the upcoming release today.

Reply:
52621